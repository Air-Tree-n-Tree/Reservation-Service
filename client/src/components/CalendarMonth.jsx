import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';

const CalendarMonth = ({ month }) => (
  <div>
    <div>{moment(month).format('MMMM')}</div>
  </div>
);

CalendarMonth.propTypes = {
  month: PropTypes.string.isRequired,
};

const mapStateToProps = ({ availability }) => ({
  minNights: availability.minNights,
  maxNights: availability.maxNights,
});

export default connect(mapStateToProps)(CalendarMonth);
