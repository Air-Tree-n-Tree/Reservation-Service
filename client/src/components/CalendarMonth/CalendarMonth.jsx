import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';

import Day from '../CalendarDay/CalendarDay';
import classes from './CalendarMonth.module.css';

export const CalendarMonth = ({ month, reservedDays }) => {
  const startOfMonthDay = moment(month).day();
  const startOfMonthDaysSince2000 = moment(month).diff(moment('2000-01-01'), 'days');
  return (
    <div className={classes.calendarContainer}>
      <div className={classes.monthLabel}>{moment(month).format('MMMM')}</div>
      <div className={classes.calendar}>
        { ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'].map((dayName) => (
          <div key={dayName}>{ dayName }</div>
        )) }

        {
          startOfMonthDay > 0
            ? <div style={{ gridColumn: `1 / ${startOfMonthDay + 1}` }} />
            : null
        }

        { reservedDays.map((dayStatus, dayOfMonth) => {
          const day = startOfMonthDaysSince2000 + dayOfMonth;
          return (
            <Day
              key={day}
              day={day}
              dayOfMonth={dayOfMonth}
              status={dayStatus}
            />
          );
        })}
      </div>
    </div>
  );
};

CalendarMonth.propTypes = {
  month: PropTypes.string.isRequired,
  reservedDays: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const mapStateToProps = ({ reservedDays, startingDay }, { month }) => {
  const startingIndex = moment(month).diff(moment('2000-01-01'), 'days') - startingDay;
  const endingIndex = startingIndex + moment(month).daysInMonth();
  return {
    startingIndex,
    reservedDays: reservedDays.slice(startingIndex, endingIndex),
  };
};

export default connect(mapStateToProps)(CalendarMonth);
