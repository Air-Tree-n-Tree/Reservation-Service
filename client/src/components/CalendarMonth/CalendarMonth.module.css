.calendar {
  display: grid;
  grid-template-columns: repeat(7, 50px);
  grid-template-rows: 30px;
  grid-auto-rows: 50px;
  place-items: center;
}
