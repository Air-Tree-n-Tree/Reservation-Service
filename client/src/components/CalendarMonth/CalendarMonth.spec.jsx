import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { CalendarMonth } from './CalendarMonth';

describe('Calender Month component', () => {
  let calendarMonth;
  beforeEach(() => {
    const dayStatuses = [];
    dayStatuses.length = moment('2020-03').daysInMonth();
    dayStatuses.fill('available');
    calendarMonth = shallow(
      <CalendarMonth
        month="2020-03"
        dayStatuses={dayStatuses}
      />,
    );
  });
  it('should render correctly', () => {
    expect(calendarMonth).toMatchSnapshot();
  });
});
