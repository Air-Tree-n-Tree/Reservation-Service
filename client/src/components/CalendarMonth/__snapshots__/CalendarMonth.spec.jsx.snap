// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calender Month component should render correctly 1`] = `
<div
  className="calendarContainer"
>
  <div
    className="monthLabel"
  >
    March
  </div>
  <div
    className="calendar"
  >
    <div
      key="Su"
    >
      Su
    </div>
    <div
      key="Mo"
    >
      Mo
    </div>
    <div
      key="Tu"
    >
      Tu
    </div>
    <div
      key="We"
    >
      We
    </div>
    <div
      key="Th"
    >
      Th
    </div>
    <div
      key="Fr"
    >
      Fr
    </div>
    <div
      key="Sa"
    >
      Sa
    </div>
    <Day
      day="2020-03-01T00:00:00-08:00"
      dayOfMonth={0}
      key="2020-03-01T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-02T00:00:00-08:00"
      dayOfMonth={1}
      key="2020-03-02T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-03T00:00:00-08:00"
      dayOfMonth={2}
      key="2020-03-03T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-04T00:00:00-08:00"
      dayOfMonth={3}
      key="2020-03-04T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-05T00:00:00-08:00"
      dayOfMonth={4}
      key="2020-03-05T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-06T00:00:00-08:00"
      dayOfMonth={5}
      key="2020-03-06T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-07T00:00:00-08:00"
      dayOfMonth={6}
      key="2020-03-07T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-08T00:00:00-08:00"
      dayOfMonth={7}
      key="2020-03-08T00:00:00-08:00"
      status="available"
    />
    <Day
      day="2020-03-09T00:00:00-07:00"
      dayOfMonth={8}
      key="2020-03-09T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-10T00:00:00-07:00"
      dayOfMonth={9}
      key="2020-03-10T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-11T00:00:00-07:00"
      dayOfMonth={10}
      key="2020-03-11T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-12T00:00:00-07:00"
      dayOfMonth={11}
      key="2020-03-12T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-13T00:00:00-07:00"
      dayOfMonth={12}
      key="2020-03-13T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-14T00:00:00-07:00"
      dayOfMonth={13}
      key="2020-03-14T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-15T00:00:00-07:00"
      dayOfMonth={14}
      key="2020-03-15T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-16T00:00:00-07:00"
      dayOfMonth={15}
      key="2020-03-16T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-17T00:00:00-07:00"
      dayOfMonth={16}
      key="2020-03-17T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-18T00:00:00-07:00"
      dayOfMonth={17}
      key="2020-03-18T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-19T00:00:00-07:00"
      dayOfMonth={18}
      key="2020-03-19T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-20T00:00:00-07:00"
      dayOfMonth={19}
      key="2020-03-20T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-21T00:00:00-07:00"
      dayOfMonth={20}
      key="2020-03-21T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-22T00:00:00-07:00"
      dayOfMonth={21}
      key="2020-03-22T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-23T00:00:00-07:00"
      dayOfMonth={22}
      key="2020-03-23T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-24T00:00:00-07:00"
      dayOfMonth={23}
      key="2020-03-24T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-25T00:00:00-07:00"
      dayOfMonth={24}
      key="2020-03-25T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-26T00:00:00-07:00"
      dayOfMonth={25}
      key="2020-03-26T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-27T00:00:00-07:00"
      dayOfMonth={26}
      key="2020-03-27T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-28T00:00:00-07:00"
      dayOfMonth={27}
      key="2020-03-28T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-29T00:00:00-07:00"
      dayOfMonth={28}
      key="2020-03-29T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-30T00:00:00-07:00"
      dayOfMonth={29}
      key="2020-03-30T00:00:00-07:00"
      status="available"
    />
    <Day
      day="2020-03-31T00:00:00-07:00"
      dayOfMonth={30}
      key="2020-03-31T00:00:00-07:00"
      status="available"
    />
  </div>
</div>
`;
